src/lib.rs

```diff
@@ -79,6 +79,14 @@ pub struct Command {
     id_map_commands: Option<(PathBuf, PathBuf)>,
 }
 
+impl Drop for Command {
+    fn drop(&mut self) {
+        for (_, fd) in &self.config.setns_namespaces {
+            unsafe { libc::close(*fd) };
```

Comment:
This should be done by making a wrapper and a Drop implementation, similarly to how is it done for passed file descriptors. This prevents the bug like in `set_namespace` that I've commented moment ago.